src/garage/sampler/worker.py

```diff
@@ -0,0 +1,215 @@
+"""Worker class used in all Samplers."""
+import abc
+from collections import defaultdict
+
+import numpy as np
+
+from garage.experiment import deterministic
+
+
+class Worker(abc.ABC):
+    """Worker class used in all Samplers."""
+
+    def __init__(self, seed, max_path_length, worker_number):
+        """Initialize a worker.
+
+        Args:
+            seed(int): The seed to use to intialize random number generators.
+            max_path_length(int): The maximum length paths which will be
```

Comment:
is `inf` a valid input here? what if I never want to stop (e.g. some off-policy envs)?